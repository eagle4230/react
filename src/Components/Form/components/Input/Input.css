.input {
    background-color: aqua;
    border-radius: 15px;
    flex-grow: 1;
    border: none;
    padding-left: 8px;
}