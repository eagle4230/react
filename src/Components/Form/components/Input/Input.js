import './Input.css';

export const Input = ({ text, setText, onChange }) => {
    return <input className="input"
        type="text"
        placeholder="  Your message..."
        value={text}
        onChange={onChange}
        data-testid="input"
    />;
}