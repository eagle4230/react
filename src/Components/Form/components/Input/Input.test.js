import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { screen } from '@testing-library/dom';
import '@testing-library/jest-dom';
import { Input } from "./Input";

describe('test_Input', () => {
    it('test_1: сравнение с снимком', () => {
        const { asFragment } = render(<Input />);
        expect(asFragment()).toMatchSnapshot();
    })
    it('test_2: renders 1', () => {
        render(<Input />);
        fireEvent.change(screen.getByRole("textbox"), {
            target: { value: "Privet" },
        });
    });

    it("test_3: renders 2", () => {
        render(<Input />);
        expect(screen.getByRole("textbox")).toBeInTheDocument();
    });

    it('test_4: вызов callback-a', () => {
        const onChange = jest.fn()
        render(<Input value="" onChange={onChange} />);
        fireEvent.change(screen.getByRole("textbox"), {
            target: { value: "JavaScript" },
        });
        expect(onChange).toHaveBeenCalledTimes(1);
    });

    it("test_5: наличие PlaceholderTexе", () => {
        render(<Input />);
        expect(screen.getByPlaceholderText("Your message...")).toBeInTheDocument();
    });
})