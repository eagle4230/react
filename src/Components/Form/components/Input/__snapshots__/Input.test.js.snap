// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test_Input test_1: сравнение с снимком 1`] = `
<DocumentFragment>
  <input
    class="input"
    data-testid="input"
    placeholder="  Your message..."
    type="text"
    value=""
  />
</DocumentFragment>
`;
